src/libstore/build.cc

```diff
@@ -1391,6 +1394,30 @@ void DerivationGoal::tryToBuild()
 {
     trace("trying to build");
 
+    /* If `build-users-group' is not empty, then we have to build as
+       one of the members of that group. */
```

Comment:
If I understand this correctly, it will try to acquire a build user even when we're doing a remote build, which is unnecessary. Maybe it's better to add a new state to the state machine, e.g. `startBuild`, with a transition from `tryToBuild` to `startBuild` if we're doing a local build.

Reply:
Oh yes, this sounds very much like something I didn't even think about.

I'll try to wrap my head around this and implement it next weekend.

Thank you again!

Reply:
Added a state and (hopefully) fixed this.